ptero_lsf/implementation/backend.py

```diff
@@ -119,3 +123,37 @@ def server_info(self):
                 'ptero_lsf.implementation.celery_app')
         result['databaseRevision'] = self.db_revision
         return result
+
+    def update_job(self, job_id, status=None):
+        job = self._get_job(job_id)
+        if job is None:
+            raise JobNotFoundError("Job %s not found" % job_id)
+
+        if status == statuses.canceled:
+            self.cancel_job(job_id, message="Job canceled via PATCH request")
+        else:
+            if status is not None:
+                job.set_status(status, message="Updated via PATCH request")
+
+        self.session.commit()
+        return job.as_dict
+
+    def cancel_job(self, job_id, message):
+        job = self._get_job(job_id)
+        if job is None:
```

Comment:
Perhaps this condition can be checked in the `_get_job()` method, and the `JobNotFoundError` exception raised there.